### ISSUE TITLE
The timeout for `wait_for_interaction` doesn't work

### ISSUE THREAD

**Issue by BlackThunder01001** [kiki7000/discord.py-components#31]
The title says it all. Seems like it loops through `wait_for` with that timeout, so each time a non relevant interaction occurs, the timeout kinda resets.

**Comment by BlackThunder01001** [comment]
For a temporary fix, I had to modify the `client.py` to count the time taken in while loop:

```
from time import time
import asyncio
#.....

        #...
        timer = timeout
        t = time()
        while True:
            timer -= time()-t
            if timer <= 5:raise asyncio.TimeoutError()
            res = await self.bot.wait_for("socket_response", check=check, timeout=timer)
            #.....
```

**Comment by kiki7000** [comment]
Fixed on 0.5.0
